ci: add rule to deny .only

boranx opened the pull request:
Fix: https://github.com/keptn/keptn/issues/8526

Kirdock reviewed (changes requested):
Hi, thanks for your contribution!

It looks like you are using `npm` instead of `yarn` or you set the wrong registry for yarn.
Could you please revert the updates in `yarn.lock` and install the dependency with `yarn add eslint-plugin-no-only-tests@3.0.0`?
FYI `yarn` is a package manager like `npm` and can be installed via `npm i -g yarn`

boranx commented:
indeed, thanks for the help! Looks like we have much less change than the previous

Kirdock commented on bridge/package.json:
```diff
@@ -64,6 +64,7 @@
     "d3": "7.6.1",
     "d3-scale": "4.0.2",
     "d3-shape": "3.1.0",
+    "eslint-plugin-no-only-tests": "3.0.0",
```
Probably my fault, but this dependency should be a dev dependency.
`yarn add -D eslint-plugin-no-only-tests@3.0.0`

boranx replied:
ah, nice catch! I've applied this in a force-push to keep the diff clean so far

Kirdock commented on bridge/.eslintrc.json:
```diff
@@ -25,13 +25,19 @@
       ],
       "rules": {
+        "no-only-tests/no-only-tests": [
+          "error", {
```
The pipeline is failing because of prettier. Could you run `prettier:fix` with yarn/npm or adapt the following lines and commit the changes?

```suggestion
          "error",
          {
```

boranx replied:
fixed, thanks!

Kirdock commented:
@boranx could you resolve the merge conflict?

boranx commented:
@Kirdock done, I've rebased the PR against the latest master and squashed the changes in a single commit.

 Here are my testing results:
 
 ```shell
 git diff
diff --git a/bridge/cypress/integration/approval.spec.ts b/bridge/cypress/integration/approval.spec.ts
index 3735dcdf9..9f6355dd7 100644
--- a/bridge/cypress/integration/approval.spec.ts
+++ b/bridge/cypress/integration/approval.spec.ts
@@ -1,6 +1,6 @@
 import { SequencesPage } from '../support/pageobjects/SequencesPage';

-describe('Test sequence screen approval', () => {
+describe.only('Test sequence screen approval', () => {
   const sequencesPage = new SequencesPage();
   const approvalId = '97b4a9a3-610a-4697-b37b-cec522a6b42f';
```

test:

```
yarn lint:check
yarn run v1.22.10
$ eslint ./

..../keptn/bridge/client/app/_views/ktb-settings-view/ktb-integration-view/ktb-modify-uniform-subscription/ktb-modify-uniform-subscription.component.ts
  363:24  warning  '_$event' is defined but never used  @typescript-eslint/no-unused-vars

..../keptn/bridge/cypress/integration/approval.spec.ts
  3:10  error  describe.only not permitted  no-only-tests/no-only-tests

✖ 2 problems (1 error, 1 warning)

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Kirdock reviewed (approved):
LGTM
@boranx thanks for your contribution!

One side note: You don't have to have a single commit in PRs, because we will squash merge it anyway
